sdk/test-utils/test-utils/test/tracing/instrumentorTracingSpanTest.spec.ts

```diff
@@ -0,0 +1,177 @@
+        assert.equal(instrumenter.currentContext(), tracingContext);
+        return 42;
+      });
+      assert.equal(result, 42);
+      assert.notEqual(instrumenter.currentContext(), tracingContext);
+    });
+    it("resets the previous context after the function returns", async function() {
+      const existingContext = instrumenter.currentContext();
+      const { tracingContext } = instrumenter.startSpan("test");
+      await instrumenter.withContext(tracingContext, async function() {
+        // no-op
+      });
+      assert.equal(instrumenter.currentContext(), existingContext);
+    });
+  });
+});
+
+describe("TestInstrumenter with MockClient", function() {
+  let instrumenter: TestInstrumenter;
+  let client: MockClientToTest;
+  beforeEach(function() {
+    instrumenter = new TestInstrumenter();
+    useInstrumenter(instrumenter);
+    client = new MockClientToTest();
+  });
+  it("starts a span and adds to startedSpans array", async function() {
+    await client.mockGetMethod();
+    assert.equal(instrumenter.startedSpans.length, 1);
+    assert.equal(instrumenter.startedSpans[0].name, "MockClientToTest.mockGetMethod");
+  });
+});
+
+describe("Test supportsTracing functionality", function() {
+  let client: MockClientToTest;
+  beforeEach(function() {
+    client = new MockClientToTest();
+  });
+  it("supportsTracing with the setMethod", function() {
+    assert.supportsTracing((options) => client.mockSetMethod({ key: "value" }, options), [
+      "MockClientToTest.mockSetMethod"
+    ]);
+    expect((options: any) => client.mockSetMethod({ key: "value" }, options)).to.supportsTracing([
+      "MockClientToTest.mockSetMethod"
+    ]);
+  });
+});
+// or something that has upgraded to core-tracing preview.14
+export class MockClientToTest {
+  private record: Record<string, string>;
+  tracingClient: TracingClient;
+
+  constructor() {
+    this.record = {};
+    this.tracingClient = createTracingClient({
+      namespace: "Microsoft.Test",
+      packageName: "@azure/test",
+      packageVersion: "foobar"
+    });
+  }
+  // const myOperationResult = await withSpan("myClassName.myOperationName", (updatedOptions) => myOperation(updatedOptions), options);
+  async mockSetMethod(record: any, options?: any) {
+    // TODO: how to pass in span options or tracing options??
+    // TODO: isn't the 2nd argument supposed to be the options for the callback??
+
+    return this.tracingClient.withSpan(
+      "MockClientToTest.mockSetMethod",
+      options,
+      (updatedOptions, span) => {
+        console.log("---Inside Callback of With span----");
+        console.log(JSON.stringify(updatedOptions.tracingOptions.tracingContext, null, 2));
+        console.log(span);
+        console.log("---Exiting Callback of With Span ---");
+        this.record = record;
+      },
+      {
+        spanKind: "consumer"
+      }
+    );
+  }
+
+  async mockGetMethod(options?: any, spanOptions?: TracingSpanOptions) {
```

Comment:
We can probably delete this method and just use the one above

Reply:
done